Builds fail on OpenBSD 7.1

It seems OpenBSD is not properly detected and it tries to find linux headers:

```
      running configure
      Using bundled libzmq
      already have bundled/zeromq
      already have platform.hpp
      checking for timer_create
      ************************************************
      ************************************************
      cc -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -g -fPIC -O2 -pipe -g -O2 -pipe -g -fPIC -c /tmp/timer_createbx4lpcl7.c -o build/temp.openbsd-7.1-amd64-cpython-39/tmp/timer_createbx4lpcl7.o
      /tmp/timer_createbx4lpcl7.c:2:5: warning: implicit declaration of function 'timer_create' is invalid in C99 [-Wimplicit-function-declaration]
          timer_create();
          ^
      1 warning generated.
      cc -pthread build/temp.openbsd-7.1-amd64-cpython-39/tmp/timer_createbx4lpcl7.o -o build/temp.openbsd-7.1-amd64-cpython-39/a.out
      ok
      ************************************************
      building 'zmq.libzmq' extension
      creating build/temp.openbsd-7.1-amd64-cpython-39/buildutils
      creating build/temp.openbsd-7.1-amd64-cpython-39/bundled
      creating build/temp.openbsd-7.1-amd64-cpython-39/bundled/zeromq
      creating build/temp.openbsd-7.1-amd64-cpython-39/bundled/zeromq/src
      c++ -pthread -std=c++11 -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -g -fPIC -O2 -pipe -g -O2 -pipe -g -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_KQUEUE=1 -DZMQ_IOTHREADS_USE_KQUEUE=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/local/include/python3.9 -c buildutils/initlibzmq.cpp -o build/temp.openbsd-7.1-amd64-cpython-39/buildutils/initlibzmq.o
      c++ -pthread -std=c++11 -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -g -fPIC -O2 -pipe -g -O2 -pipe -g -fPIC -DZMQ_HAVE_CURVE=1 -DZMQ_USE_TWEETNACL=1 -DZMQ_USE_KQUEUE=1 -DZMQ_IOTHREADS_USE_KQUEUE=1 -DZMQ_POLL_BASED_ON_POLL=1 -Ibundled/zeromq/include -Ibundled -I/usr/local/include/python3.9 -c bundled/zeromq/src/address.cpp -o build/temp.openbsd-7.1-amd64-cpython-39/bundled/zeromq/src/address.o
      In file included from bundled/zeromq/src/address.cpp:38:
      bundled/zeromq/src/tipc_address.hpp:43:10: fatal error: 'linux/tipc.h' file not found
      #include <linux/tipc.h>
               ^~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/c++' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyzmq
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,4 @@
-# 3.28 needed for EXCLUDE_FROM_ALL
-cmake_minimum_required(VERSION 3.28)
+cmake_minimum_required(VERSION 3.14...3.28)
 project(${SKBUILD_PROJECT_NAME} LANGUAGES C CXX)
 set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
 
@@ -280,7 +279,6 @@ if (ZMQ_PREFIX STREQUAL "bundled")
   FetchContent_Declare(bundled_libzmq
     URL ${PYZMQ_LIBZMQ_URL}
     PREFIX ${BUNDLE_DIR}
-    EXCLUDE_FROM_ALL
   )
   FetchContent_MakeAvailable(bundled_libzmq)
 
@@ -364,4 +362,8 @@ else()
 endif()
 
 target_include_directories(${ZMQ_EXT_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/zmq/utils")
-install(TARGETS ${ZMQ_EXT_NAME} DESTINATION "${ZMQ_BACKEND_DEST}")
+install(TARGETS ${ZMQ_EXT_NAME} DESTINATION "${ZMQ_BACKEND_DEST}" COMPONENT pyzmq)
+
+# add custom target so we exclude bundled targets from installation
+# only need this because the extension name is different for cff/cython
+add_custom_target(pyzmq DEPENDS ${ZMQ_EXT_NAME})

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -272,6 +272,24 @@ if (ZMQ_PREFIX STREQUAL "bundled")
   set(BUILD_TESTS OFF)
   set(BUILD_SHARED OFF)
   set(BUILD_STATIC ON)
+
+  if(NOT MSVC)
+    # backport check for kqueue, which is wrong in libzmq 4.3.5
+    # libzmq's cmake will proceed with the rest
+    # https://github.com/zeromq/libzmq/pull/4659
+    include(CheckCXXSymbolExists)
+    set(POLLER
+      ""
+      CACHE STRING "Choose polling system for I/O threads. valid values are
+    kqueue, epoll, devpoll, pollset, poll or select [default=autodetect]")
+    if(POLLER STREQUAL "")
+      check_cxx_symbol_exists(kqueue "sys/types.h;sys/event.h;sys/time.h" HAVE_KQUEUE)
+      if(HAVE_KQUEUE)
+        set(POLLER "kqueue")
+      endif()
+    endif()
+  endif()
+
   if(MSVC)
     set(API_POLLER "select" CACHE STRING "Set API Poller (default: select)")
   endif()

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -67,7 +67,11 @@ readme = { file = "README.md" }
 
 [tool.scikit-build]
 wheel.packages = ["zmq"]
-cmake.version = ">=3.28"
+# 3.14 for FetchContent_MakeAvailable
+cmake.version = ">=3.14"
+# only build/install the pyzmq component
+cmake.targets = ["pyzmq"]
+install.components = ["pyzmq"]
 
 [tool.autoflake]
 ignore-init-module-imports = true

